testdata/metric/client_request_total.yaml.tmpl

```diff
@@ -31,21 +31,21 @@ metric:
   - name: destination_version
     value: v1
   - name: destination_service
-    value: 127.0.0.1:{{ .Vars.ClientPort }}
+    value: server.default.svc.cluster.local
   - name: destination_canonical_service
     value: ratings
   - name: destination_canonical_revision
     value: version-1
   - name: destination_service_name
-    value: 127.0.0.1:{{ .Vars.ClientPort }}
+    value: server
   - name: destination_service_namespace
     value: default
   - name: request_protocol
     value: http
```

Comment:
This looks like an issue: it should be `grpc`. Maybe file an issue?

Reply:
In the grpc test we only verify server side metrics. I updated the test to also verify client side metrics and now this `request_protocol` field is filled selectively based on presence of GrpcResponseCode.